### PR TITLE
Autofix group problems

### LINKED ISSUE #4455: Autofix group problems in the inspector

### PR DESCRIPTION
**Pull request by ruggi**
Fixes #4455 

**Problem:**

Group with invalid configurations show up in the navigator and can be fixed in code or with the inspector… if you know where to look.

An example with broken groups: 
- master: https://utopia.pizza/p/0df967c7-sprinkle-base/
- this PR: https://utopia.pizza/p/0df967c7-sprinkle-base//?branch_name=fixeat-inspector-invalid-groups

**Fix:**

This PR adds an autofix button in the inspector that fixes invalid group configurations in place (if possible). The button is positioned in the contract section of the inspector.

Screenshot 2023-11-01 at 3 49 21 PM

When clicking it, any selected groups or group children will be adjusted and trued-up.

Additionally, labels for offending props (tlbr, size, constraints) are shown in red in the inspector inputs and marked as `Invalid`.

https://github.com/concrete-utopia/utopia/assets/1081051/0c78a3c6-5833-4270-ae43-564bfd94ae34

### CODE REVIEW

File: editor/src/core/performance/performance-regression-tests.spec.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ describe('React Render Count Tests -', () => {
 
     const renderCountAfter = renderResult.getNumberOfRenders()
     // if this breaks, GREAT NEWS but update the test please :)
-    expect(renderCountAfter - renderCountBefore).toMatchInlineSnapshot(`734`)
+    expect(renderCountAfter - renderCountBefore).toMatchInlineSnapshot(`714`)
```
Comment (Rheeseyb) [inline editor/src/core/performance/performance-regression-tests.spec.tsx:68]: Some went up and some went down?

Comment (ruggi) [reply editor/src/core/performance/performance-regression-tests.spec.tsx:68]: indeed!

File: editor/src/components/inspector/constraints-section.tsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ export const ConstraintsSection = React.memo(() => {
     <React.Fragment>
       <InspectorSubsectionHeader>
         <FlexRow
-          data-testId={InspectorSectionConstraintsTestId}
+          data-testid={InspectorSectionConstraintsTestId}
```
Comment (balazsbajorics) [inline editor/src/components/inspector/constraints-section.tsx:86]: our national nightmare is over :D

**Review by balazsbajorics — APPROVED** [review]

File: editor/src/components/inspector/constraints-section.tsx (hunk 2 above)
Comment (ruggi) [reply editor/src/components/inspector/constraints-section.tsx:86]: hear hear 🎺

**Review by Rheeseyb — APPROVED** [review]

**Pull request merged**